### DISCUSSION TITLE
MQTT TLS Configuration issue

### DISCUSSION THREAD

**Discussion by mohitvaid** [syslog-ng/syslog-ng#4008] [Q&A] [answered]
Hi Team,

I am observing _**ecdh-curve-list**_ is not getting considered for MQTT communication with TLS, syslog-ng v3.36.

I am getting conf file parsing error.

Is this option intentionally left out for MQTT configuration? Based on https://github.com/syslog-ng/syslog-ng/discussions/3983, I hope this change supports client-algo selection for MQTT TLS as well.

I am observing `crl, ecdh-curve-list, signature-algs` are also not supported for MQTT TLS,

My configuration:

```
destination d_mqtt_network_1 {
        mqtt(
        address("ssl://10.108.172.117:21000")
        topic("alpha")
        fallback-topic("syslog/fallback")
        client-id("test")
        tls(
            ca-file("syslogCACert.pem")
            key-file("mqttClientKey_1.pem")
            cert-file("mqttClientCert_1.pem")

            ecdh-curve-list("prime256v1:secp384r1")
            peer-verify(yes)
        )
        );
};

filter f_mqtt_whitelist_1 { in-list("syslog.list", value("PROGRAM")); };

log {
        source(s_local);
        filter(f_mqtt_whitelist_1);
        destination(d_mqtt_network_1);
};
```

But with general syslog-tls the `ecdh-curve-list("prime256v1:secp384r1")` is getting picked correctly.

Kindly clarify.

**Comment by MrAnno** [comment, marked as answer]
Adding `ecdh-curve-list()` for the MQTT module needs support from the paho.mqtt.c side.
Let's see what happens with eclipse/paho.mqtt.c#1226
